### PR TITLE
TCCP: JSON detail view + nicer enum display

### PR DESCRIPTION
**Pull request by chosak**
This commit refactors the TCCP detail view so that it is accessible via JSON, at, for example with the current dataset at http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/baxter-credit-union-travel-rewards-visa-signature/?format=json

This work was done in service of making it easier to debug this view and to provide a more consistent framework for converting TCCP enum values to user-friendly strings on the frontend.

The list view still also works as JSON, at http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/?format=json

This change includes two places where we now use more user-friendly strings: card rewards, and card state names.

To see the more concise rewards labels on the list view, visit http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/ and notice the shorter labels under "Rewards".

To see the nicer labels for both rewards and states, visit a card detail page with both of these, for example http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/banner-bank-banner-bank-trurewards-mastercard/

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)

### CODE REVIEW

File: cfgov/tccp/jinja2/tccp/card.html [hunk 1]
```diff
@@ -30,12 +30,22 @@ <h2>Application requirements</h2>
...
+            {% endif -%}
+
+            {% if not state_limitations %}
                 Available to anyone in the United States
             {% else %}
                 Available only to people who live in
                 {{ 'specific counties in ' if card.pertains_to_specific_counties }}
-                {{ card.state_limitations | join(", ") }}
+                {% for state in state_limitations %}
+                    {{ state_lookup[state] }}{% if not loop.last %},{% endif %}
```
Comment (niqjohnson) [inline cfgov/tccp/jinja2/tccp/card.html:47]: I think to cleanly handle cases with just two states and to get our Oxford comma in, this should work:

```suggestion
                    {{ state_lookup[state] -}}
                    {{- ', ' if not loop.last and loop.length > 2 }}
                    {{ 'and ' if loop.index == loop.length - 1}}
```

**Review by niqjohnson — APPROVED** [review]
Working great for me, though @contolini might want to look a the actual code as it was mostly over my head. I did suggest one change to make the list of states read a little more naturally for two states (and for Oxford comma lovers!), but I can make that change in my card details cleanup if it makes more sense there.

File: cfgov/tccp/jinja2/tccp/card.html (hunk 1 above)
Comment (chosak) [reply cfgov/tccp/jinja2/tccp/card.html]: Thanks @niqjohnson, that's a great suggestion. I've committed your change with some additional whitespace collapsing in 3ab7617.

**Review by contolini — APPROVED** [review]
:shipit:

**Pull request merged**